src/examples/fake_device_demo.rb

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path('../../lib', __FILE__))
+
+require "storage/fake_probing.rb"
+require "storage/fake_device_factory.rb"
+
+FILENAME = "fake-devicegraphs"
+
+fake_probing = Yast::Storage::FakeProbing.new
+devicegraph = fake_probing.devicegraph
+factory = Yast::Storage::FakeDeviceFactory.new(devicegraph)
+factory.load_yaml_file("#{FILENAME}.yml")
+
+fake_probing.to_probed
+puts("Disks:")
+probed = Yast::Storage::StorageManager.instance.probed
+fake_probing.dump_disks(probed)
```

Comment:
Running this in my system doesn't seem to trigger the hardware probing, although I could be wrong (or I could have misunderstood you):
```ruby
require "storage"
graph = Storage::Devicegraph.new
Storage::Disk.create(graph, "/dev/sdaX")
puts graph.to_s
```